irohad/torii/impl/command_service.cpp

```diff
@@ -249,4 +203,59 @@ namespace torii {
     return grpc::Status::OK;
   }
 
+  void CommandService::checkCacheAndSend(
+      boost::optional<iroha::protocol::ToriiResponse> &resp,
+      grpc::ServerWriter<iroha::protocol::ToriiResponse> &response_writer)
+      const {
+    if (resp) {
+      if (resp->tx_status()
+              == iroha::protocol::TxStatus::STATELESS_VALIDATION_FAILED
+          or resp->tx_status()
+              == iroha::protocol::TxStatus::STATEFUL_VALIDATION_FAILED
+          or resp->tx_status() == iroha::protocol::TxStatus::NOT_RECEIVED
+          or resp->tx_status() == iroha::protocol::TxStatus::COMMITTED) {
+        response_writer.WriteLast(*resp, grpc::WriteOptions());
+        return;
+      }
+      response_writer.Write(*resp);
+    }
+  }
+
+  iroha::protocol::TxStatus CommandService::convertStatusToProto(
+      iroha::model::TransactionResponse::Status &status) {
+    iroha::protocol::TxStatus proto_status;
+    switch (status) {
```

Comment:
better to use map instead.

Reply:
I do not think that std::transform will be good here: we'll just move this switch inside lambda and also add several lines of boilerplate code. Or may be you mean something else?